gufe/transformations/transformation.py

```diff
@@ -0,0 +1,332 @@
+        self._mapping = mapping
+        self._name = name
+
+        self._protocol = protocol
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(initial={self.initial}, final={self.final}, protocol={self.protocol})"
+
+    @property
+    def initial(self):
+        """The starting `ChemicalSystem` for the transformation.
+
+        """
+        return self._initial
+
+    @property
+    def final(self):
+        """The ending `ChemicalSystem` for the transformation.
+
+        """
+        return self._final
+
+    @property
+    def protocol(self):
+        """The protocol for sampling the transformation to derive free energy
+        differences between `initial` and `final` `ChemicalSystem`s.
+
+        """
+        return self._protocol
+
+    @property
+    def mapping(self):
+        """The mapping between atoms in `initial` to `final` `ChemicalSystem`s for
+        the transformation.
+
+        """
+        return self._mapping
+
+    @property
+    def name(self):
+        """User-specified for the transformation; used as part of its hash.
+
+        """
+        return self._name
+
+    def __lt__(self, other):
+        return hash(self) < hash(other)
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        if self._name != other.name:
+            return False
+        if self._mapping != other.mapping:
+            return False
+        if self._final != other.final:
+            return False
+        if self._initial != other.initial:
+            return False
+
+        return True
+
+    def __hash__(self):
+        return hash(
+            (
+                self._initial,
+                self._final,
+                self._mapping,
+                self._name,
+            )
+        )
+
+    # TODO: broken without a `Protocol` registry of some kind
+    # should then be changed to use the registry
+    def to_dict(self) -> dict:
+        return {
+            "initial": self.initial.to_dict(),
+            "final": self.final.to_dict(),
+            "protocol": self.protocol.to_dict(),
+        }
+
+    # TODO: broken without a `Protocol` registry of some kind
+    # should then be changed to use the registry
+    @classmethod
+    def from_dict(cls, d: dict):
+        return cls(
+                initial=ChemicalSystem.from_dict(d['initial']),
+                final=ChemicalSystem.from_dict(d['final']),
+                protocol=Protocol.from_dict(d['protocol'])
+                )
+
+    def create(
+            self, 
+            settings: Optional["ProtocolSettings"] = None) -> ProtocolDAG:
+        """Returns a generator of WorkUnits.
+
+        Parameters
+        ----------
+        settings
+            Override level 3 settings with a ProtocolSettings.
+
+        """
+        # TODO: need an update path for level-3 settings overrides
+        if settings is not None:
+            self.protocol.__class__(settings=settings)
```

Comment:
Related to the comment above wrt settings, but I think this is mutating the transformation which might break the Network (depending on assumptions on how edges hash/behave)